FreeRTOS/Makefile

```diff
@@ -0,0 +1,366 @@
+AS = $(CROSS_COMPILE)gcc -x assembler-with-cpp
+CP = $(CROSS_COMPILE)objcopy
+SZ = $(CROSS_COMPILE)size
+endif
+HEX = $(CP) -O ihex
+BIN = $(CP) -O binary -S
+ 
+#######################################
+# CFLAGS
+#######################################
+# cpu
+CPU = -mcpu=cortex-m4
+
+# fpu
+FPU = -mfpu=fpv4-sp-d16
+
+# float-abi
+FLOAT-ABI = -mfloat-abi=hard
+
+# mcu
+MCU = $(CPU) -mthumb $(FPU) $(FLOAT-ABI)
+
+# macros for gcc
+# AS defines
+AS_DEFS = 
+
+# C defines
+C_DEFS =  \
+-DUSE_HAL_DRIVER \
+-DSTM32F407xx \
+-D_TIMEVAL_DEFINED
+
+# AS includes
+AS_INCLUDES =  \
+-IInc
+
+# C includes
+C_INCLUDES =  \
+-IInc \
+-IDrivers/STM32F4xx_HAL_Driver/Inc \
+-IDrivers/STM32F4xx_HAL_Driver/Inc/Legacy \
+-IMiddlewares/Third_Party/FreeRTOS/Source/include \
+-IMiddlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2 \
+-IMiddlewares/Third_Party/FreeRTOS/Source/portable/GCC/ARM_CM4F \
+-IDrivers/CMSIS/Device/ST/STM32F4xx/Include \
+-IDrivers/CMSIS/Include \
+-IDrivers/CMSIS/Include \
+-IMiddlewares/Third_Party/LwIP/src/include \
+-IMiddlewares/Third_Party/LwIP/system \
+-IMiddlewares/Third_Party/LwIP/src/include/netif/ppp \
+-IMiddlewares/Third_Party/LwIP/src/include/lwip \
+-IMiddlewares/Third_Party/LwIP/src/include/lwip/apps \
+-IMiddlewares/Third_Party/LwIP/src/include/lwip/priv \
+-IMiddlewares/Third_Party/LwIP/src/include/lwip/prot \
+-IMiddlewares/Third_Party/LwIP/src/include/netif \
+-IMiddlewares/Third_Party/LwIP/src/include/posix \
+-IMiddlewares/Third_Party/LwIP/src/include/posix/sys \
+-IMiddlewares/Third_Party/LwIP/system/arch
+
+
+C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/include
+C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/include/FreeRTOS_POSIX
+C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/include/private
+C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/FreeRTOS-Plus-POSIX/include
+C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/FreeRTOS-Plus-POSIX/include/portable
+C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/FreeRTOS-Plus-POSIX/include/portable/st/stm32l475_discovery
+C_INCLUDES += -I$(PROJECTFOLDER)/micro_xrce_dds_install/include
+C_INCLUDES += -I$(PROJECTFOLDER)
+
+# C_INCLUDES += -I$(PROJECTFOLDER)/micro_xrce_dds_install/include
+# C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/include/FreeRTOS_POSIX
+# C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/FreeRTOS-Plus-POSIX/include
+# C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/include/private
+# C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/FreeRTOS-Plus-POSIX/include/portable/st/stm32l475_discovery
+# C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/FreeRTOS-Plus-POSIX/include/portable/
+# C_INCLUDES += -I$(PROJECTFOLDER)/Middlewares/Third_Party/FreeRTOS_POSIX/include
+# C_INCLUDES += -I$(PROJECTFOLDER)

```

Comment:
Please, remote it in case it is not necessary.

Reply:
Done